scripts/example_script.py

```diff
@@ -0,0 +1,241 @@
+    print ('Request Error:\nstatus:%s\nresponse%s\nurl:%s'
+           '\nmethod:%s ' % (err.status, err.responseText, err.url, err.method))
+
+
+def recurse_get_resource(client, parent_folder, resource_type,
+                         resource_list, parent_is_collection=False):
+    # now get all folders
+    folders = None
+    folder_id_list = None
+    if parent_is_collection:
+        try:
+            folders = client.listFolder(parent_folder,
+                                        parentFolderType='collection')
+        except HttpError as err:
+            print_http_error(err)
+            exit()
+        folder_id_list = get_field(folders, '_id')
+        if resource_type is 'folder':
+            resource_list.extend(folders)
+
+    else:
+
+        try:
+            folders = client.listFolder(parent_folder)
+        except HttpError as err:
+            print_http_error(err)
+            return
+        folder_id_list = get_field(folders, '_id')
+
+        if resource_type is 'item':
+            try:
+                data = client.listItem(parent_folder)
+
+                resource_list.extend(data)
+            except HttpError as err:
+                print_http_error(err)
+                exit()
+        elif resource_type is 'folder':
+            resource_list.extend(folders)
+        else:
+            return
+    for folder_id in folder_id_list:
+        recurse_get_resource(client, folder_id, resource_type, resource_list)
+
+
+def print_field(data, str_key):
+    for i in range(len(data)):
+        print(data[i][str_key])
+
+
+def get_field(data, str_key):
+    return [i[str_key] for i in data]
+
+
+def turn_into_a_large_image(girder_client, file_id):
+    # first check if the file is already a large image
+    item_data = None
+    try:
+        item_data = gc.getItem(file_id)
+    except HttpError as err:
+        print_http_error(err)
+        print('bad item id ')
+    # if the item is already a large image return
+
+    if 'largeImage' in item_data:
+        return
+    try:
+        girder_client.post('item/%s/tiles' % file_id)
+    except HttpError as err:
+        print_http_error(err)
+
+
+def isEmpty(girder_client, folder):
+    item_list = []
+    recurse_get_resource(girder_client, folder, 'item', item_list)
+    if len(item_list) == 0:
+        return True
+    else:
+        return False
+girder_location = 'http://localhost:8080/api/v1'
+
+collection_name = 'DG TCGA'
+collection_id = None
+request_url = 'resource/lookup?path=collection/%s' % (collection_name)
+collection_data = None
+
+new_folder_name = 'GBM'
+new_folder_desc = 'orginizing patient data'
+new_folder_id = None
+
+img_type = '.jpeg'
+
+# login to girder, you will be prompted for credentials
+gc = girder_client.GirderClient(apiUrl=girder_location)
+gc.authenticate(interactive=True)
+
+# get the id of the desired collection
+try:
+    collection_data = gc.getResource(request_url)
+except HttpError as err:
+    print_http_error(err)
```

Comment:
Once you put this script in a `main` function, you should wrap the entire call to main within a `try...except HttpError...` block, to avoid having to duplicate the calls to `print_http_error` and `exit`. However, do make a code comment to note that the calls to `gc` are the expected source of `HttpError`s.